doc/dev.md

```diff
@@ -24,68 +24,50 @@ Pull request
-2. The `pem` file containing the private key associated to the app
-3. A comma separated list of GitHub accounts to allow - this could start out as just your GitHub account
-4. A capability file for submitting jobs to a cluster, in this case the main ocaml-ci cluster as documented in https://github.com/ocurrent/ocluster#admin
-5. The app webhook secret that is used to authenticate to the app
+1. Under `services`, then `service`, change the argument to `--github-app-id` to the GitHub App ID of the app you created
+2. Under the same `service` tag, change the argument to `--github-account-allowlist` to a comma-separated list of GitHub accounts to allow—this could start out as just your GitHub account
+3. Under `secrets`, set `ocaml-ci-github-key` to the `pem` file containing the private key associated to the app, and `ocaml-ci-webhook-secret` to a file containing the webhook secret that is used to authenticate to the app
 
+Build the docker containers for `service` and `web`:
 ```
-dune exec -- ocaml-ci-service \
-  --github-app-id <your-github-app-id> \
-  --github-private-key-file <path-to/private-key.pem> \
-  --github-account-allowlist <your-github-username> \
-  --submission-service <path-to-the-submission-capability-file> \
-  --github-webhook-secret-file <path-to-the-app-secret> \
-  --capnp-listen-address tcp:127.0.0.1:9001
-  --migration-path "$PWD/migrations"
+docker build -t ocaml-ci-service -f Dockerfile .
+docker build -t ocaml-ci-web -f Dockerfile.web .
 ```
 
-This will generate a capability file. See the logs for `Wrote capability reference to "./capnp-secrets/ocaml-ci-admin.cap"`
-
-You should see the admin site on `localhost:8080`
-
-### Running the GitLab pipeline locally
-
-You will need the following:
-
-1. The GitLab API token with permissions to the repositories to build
-2. GitLab secret associated with webhooks
-3. A capability file for submitting jobs to a cluster, in this case the main ocaml-ci cluster as documented in https://github.com/ocurrent/ocluster#admin
-
-``` shell
-dune exec -- ocaml-ci-gitlab \
-  --gitlab-token-file <your-gitlab-token> \
-  --gitlab-webhook-secret-file <your-gitlab-secret> \
-  --submission-service <path-to-the-submission-capability-file> \
-  --capnp-listen-address tcp:127.0.0.1:9800
-  --migration-path "$PWD/migrations"
+Create a file `/etc/caddy/Caddyfile` containing:
+```
+{
+	log default {
+		level WARN
+	}
+}
+
+http://localhost:8100 {
+	reverse_proxy service:8080
+}
+
+http://localhost {
+	reverse_proxy web:8090
+}
 ```
 
-This will generate a capability file. See the logs for `Wrote capability reference to "./capnp-secrets/ocaml-ci-gitlab-admin.cap"`
-
-### Running the web client locally
-
-Using the capability file written out by the service, run the web client as follows:
-
+You can then start the services with:
 ```
-dune exec -- ocaml-ci-web \
-  --backend ./capnp-secrets/ocaml-ci-admin.cap \
-  --gitlab-backend ./capnp-secrets/ocaml-ci-gitlab-admin.cap
+docker compose up
```

Comment:
SECRET is a directory that contains the following files: `private-key.pem` and `webhook-secret` the command could be like
```
$ APP_ID=244548 ALLOW_LIST="\'benmandrew\'" SECRETS=secrets_dir  \ 
  docker-compose -f  docker-compose.yml build
```